src/stcal/jump/jump.py

```diff
@@ -262,3 +299,131 @@ def detect_jumps(frames_per_group, data, gdq, pdq, err,
 
     # Return the updated data quality arrays
     return gdq, pdq
+
+
+def flag_large_events(gdq, jump_flag, sat_flag, min_sat_area=1,
+                      min_jump_area=6,
+                      expand_factor=1.9, use_ellipses=False,
+                      sat_required_snowball=True):
+    """
+    This routine controls the creation of expanded regions that are flagged as jumps. These are called
+    snowballs for the NIR and are almost always circular with a saturated core. For MIRI they are better
+    fit with ellipses.
+
+    :param gdq: The group DQ cube for all integrations
+    :param jump_flag: The bit value that represents jump
+    :param sat_flag:  The bit value that represents saturation
+    :param min_sat_area:  The minimum area of saturated pixels within the jump circle to trigger the
+                        creation of a snowball.
+    :param min_jump_area:  The minimum area of a contour to cause the creation of the minimum
+                        enclosing ellipse or circle
+    :param expand_factor: The factor that increases the size of the snowball or enclosing ellipse.
+    :param use_ellipses:  Use ellipses rather than circles (better for MIRI)
+    :param sat_required_snowball: Require that there is a saturated pixel within the radius of the jump
+                        circle to trigger the formation of a snowball.
+    :return: none
+    """
+    for integration in range(gdq.shape[0]):
```

Comment:
Interesting: so this whole routine relies only on information contained in the previously set DQ flags and not the actual pixel data. (nothing wrong with that - just find it interesting!)

Reply:
The combination of the SAT flag and the JUMP flag provides enough information to correctly flag over 90% of the large events. In the future, I think we could increase the fidelity of the flagging using the magnitude of the jumps but I wanted to start simple since this is a major change.